lib/util/common.js

```diff
@@ -12,10 +12,23 @@ function addHook (fastify, pluginOptions) {
   const sharedSchemasMap = new Map()
 
   fastify.addHook('onRoute', (routeOptions) => {
-    // we need to skip HEAD route
-    // since it may automatically added by fastify
-    // and will conflict with the existing route
-    if (routeOptions.method === 'HEAD') return
+    // we need to skip the HEAD route if there is
+    // already one route with the same operationId.
+    // since https://github.com/fastify/fastify/pull/2826 head routes 
+    // are created by default in fastify
+    if (
+      routeOptions.method === "HEAD" &&
+      routes.find(
+        ({  schema }) =>
+          (schema != null &&
+            schema.operationId != null &&
+            routeOptions.schema != null &&
+            routeOptions.schema.operationId != null &&
+            schema.operationId === routeOptions.schema.operationId) 
+      ) != null
```

Comment:
Can we not use !== ?

Reply:
Ah yes of course, but that means also I have to use `undefined` instead of `null`. Are you fine with `!find(...)`?